bouncer/sentry.py

```diff
@@ -0,0 +1,31 @@
+"""Report exceptions to Sentry/Raven."""
+import raven
+
+
+def raven_tween_factory(handler, _):
+    """Return a tween that reports uncaught exceptions to Sentry."""
+    def raven_tween(request):
+        """Report uncaught exceptions to Sentry."""
+        try:
+            return handler(request)
+        except:
+            request.raven.captureException()
+            raise
+
+    return raven_tween
+
+
+def raven_client(request):
+    """Return the raven.Client object."""
+    request.add_finished_callback(
+        lambda request: request.raven.context.clear(),
+    )
+
+    return raven.Client(
+        dsn=request.registry.settings.get("sentry_dsn"),
+    )
+
+
+def includeme(config):
+    config.add_request_method(raven_client, name="raven", reify=True)
+    config.add_tween("bouncer.sentry.raven_tween_factory")
```

Comment:
A call to `add_tween` without specifying `under` or `over` hints means:

> Specifying neither over nor under is equivalent to specifying under=INGRESS.

(From [the Pyramid docs](http://docs.pylonsproject.org/projects/pyramid/en/latest/api/config.html))

That means that the ordering of tweens here will be:

```
INGRESS
raven_tween_factory
EXCVIEW
MAIN
```

Which in turn means that the pyramid excview will catch all the exceptions before they get to Raven. You probably want to specify `under=EXCVIEW` here.